pkg/provider/resource_fareplica.go

```diff
@@ -277,6 +280,38 @@ func (r *FAReplicaResource) Schema(ctx context.Context, req resource.SchemaReque
 				ElementType:   types.StringType,
 				PlanModifiers: []planmodifier.Set{setplanmodifier.UseStateForUnknown()},
 			},
+			"transparent_data_encryption": schema.SingleNestedAttribute{
```

Comment:
TDE can only be set for faraway-replica if source cluster is TDE enabled, need to validate the condition or need to rely on API error

Reply:
I was thinking to let them set transparent_data_encryption.key_id and they should get an API error if it fails.

Reply:
in that case we need to check if the API error gives enough information or else we should show some graceful error relateable to context

Reply:
Ok. I did do a test it seems like faraway-replica is broken. I think I fixed it in v1 I recall, so I think I will test it again after v1 is merged rather than trying to fix it in this PR